packages/beacon-node/src/util/dataColumns.ts

```diff
@@ -142,19 +142,19 @@ function computeSubnetForDataColumn(columnIndex: ColumnIndex): number {
  * Calculate the number of custody groups the node should subscribe to based on the node's effective balance
  *
  * SPEC FUNCTION
- * https://github.com/ethereum/consensus-specs/blob/dev/specs/fulu/validator.md#validator-custody
+ * https://github.com/ethereum/consensus-specs/blob/v1.6.0-alpha.3/specs/fulu/validator.md#validator-custody
  */
 export function getValidatorsCustodyRequirement(
-  state: CachedBeaconStateAllForks,
-  validatorIndices: ValidatorIndex[],
-  config: ChainForkConfig
+  config: ChainForkConfig,
+  stateValidators: ListCompositeTreeViewDU<typeof ssz.phase0.Validator>,
+  validatorIndices: ValidatorIndex[]
```

Comment:
can/should we just pass in a list of effective balances (and config)?

Reply:
depends on how close we wanna keep it to the [spec function](https://github.com/ethereum/consensus-specs/blob/0da5c4b5bb0caa80411e34ec77d36b3166adc27d/specs/fulu/validator.md?plain=1#L123-L132), it's already different though because we no longer pass the state

```python
def get_validators_custody_requirement(
    state: BeaconState, validator_indices: Sequence[ValidatorIndex]
) -> uint64:
    total_node_balance = sum(
        state.validators[index].effective_balance for index in validator_indices
    )
    count = total_node_balance // BALANCE_PER_ADDITIONAL_CUSTODY_GROUP
    return min(max(count, VALIDATOR_CUSTODY_REQUIREMENT), NUMBER_OF_CUSTODY_GROUPS)
```

Reply:
yea i don't think that matters very much here

Reply:
tbh i dislike the spec functions that require a whole state to do something (and then just use one field).

Reply:
yeah can do the refactoring, if fine with it either way